pages/writing.md

```diff
@@ -45,23 +45,131 @@ juliaup status
+For functions, types, and variables, the query fetches things such as documentation, type fields and supertypes, and in which file the object is defined.
+<!-- How do you do syntax highlighting for the Julia REPL? -->
+```
+help?> Int
+search: Int Int8 Int64 Int32 Int16 Int128 Integer intersect intersect!
+
+  Int64 <: Signed
+
+
+  64-bit signed integer type.
+```
+
+For unicode symbols, the query will return how to type the symbol in the REPL, which is useful when you copy-paste a symbol in without knowing its name, and fetch information about the object the symbol is bound to, just as above.
+
+### Pkg mode (`]`)
+Pkg mode is for managing environments and packages that is based on the Rust package manager "cargo".
+By pressing `]` in Julian mode, you can `add`, `update` (or `up`) and `remove` (or `rm`) packages, `activate` different local or global environments, and get the `status` (or `st`) of your current environment.
+
+More detail on using Pkg.jl and Pkg mode, see the [#Package-Management] section or the [#Setup] section of the [Sharing Julia code](./sharing.md#setup) post.
+
+<!-- Check whether the second link is needed after writing both sections. -->
+
+### Shell mode(`;`)
+Shell mode: Functions as a terminal inside Julia, you can also execude shell commands from Julian mode.
+Here's an example
+```julia
+an_example()
+```
+
+<!-- More needs to be written about the `edit` functionality of the REPL, I should talk to someone who does REPL-driven development. Miguel? -->
+
 * [VSCode](https://code.visualstudio.com/) / [VSCodium](https://vscodium.com/) + [Julia VSCode extension](https://www.julia-vscode.org/)
 * [emacs](https://www.gnu.org/software/emacs/) / [vim](https://www.vim.org/) / other IDEs + [JuliaEditorSupport](https://github.com/JuliaEditorSupport)
 * [Jupyter](https://jupyter.org/) / [IJulia.jl](https://github.com/JuliaLang/IJulia.jl)
 * [Pluto.jl](https://plutojl.org/)
 
 ## Running code
 
-* [REPL](https://docs.julialang.org/en/v1/stdlib/REPL/)
-* [Revise.jl](https://github.com/timholy/Revise.jl)
-* [running in VSCode](https://www.julia-vscode.org/docs/stable/userguide/runningcode/)
-* startup file
+All executed Julia code ends up in a REPL[^1], but how it gets there can vary greatly.
+
+[^1]: Unless you're compiling binaries with [StaticTools.jl](https://github.com/brenhinkeller/StaticTools.jl).
+
+<!-- The VSCode is important to write in tandem with or after the Development environments part is written to avoid overlap. -->
+Most people develop Julia using a tool that allows for [interactive development](https://en.wikipedia.org/wiki/Interactive_programming).
+In VScode, you can [run code](https://www.julia-vscode.org/docs/stable/userguide/runningcode/) in the following useful ways:
+* For interactive development, the standard way to use Julia, you can use hotkeys to run selections, lines, sections, or whole files in the currently open REPL.
```

Comment:
Not clear to a beginner what a hotkey is. Mention the universal shortcut `Ctrl + Shift + P` for the command pane?

Reply:
I think hotkey is a common enough term that I feel comfortable using it. 3rd opinion?

That said, this part's not written yet.

Reply:
FWIW, the Julia VSCode docs use the term ["keyboard shortcuts"](https://www.julia-vscode.org/docs/stable/userguide/keybindings/#Keyboard-shortcuts).